test/functions_test.rb

```diff
@@ -117,6 +133,23 @@ def test_function_that_returns_a_sass_value
       CSS
     end
 
+    def test_function_that_returns_a_sass_map
+      assert_sass <<-SCSS, <<-CSS
+        $my-map: returns-sass-map();
+        div { background: map-get( $my-map, color ); }
+      SCSS
+        div { background: black; }
+      CSS
+    end
+
+    def test_function_that_takes_a_sass_map
+      assert_sass <<-SCSS, <<-CSS
+        div { background-color: map-get( returns-arg(( dark: black, light: white )), dark ); }
```

Comment:
Rather than just returning the map, it would be nice to test that the map values are usable within the ruby function.  So perhaps you could pass in a map with a string, number, etc, and ensure that within the ruby function they are usable and have the correct types.